PathPlanning/RRT/sobol/sobol.py

```diff
@@ -0,0 +1,917 @@
+"""
+  Licensing:
+    This code is distributed under the MIT license.
+
+  Authors:
+    Original FORTRAN77 version of i4_sobol by Bennett Fox.
+    MATLAB version by John Burkardt.
+    PYTHON version by Corrado Chisari
+
+    Original Python version of is_prime by Corrado Chisari
+
+    Original MATLAB versions of other functions by John Burkardt.
+    PYTHON versions by Corrado Chisari
+
+    Original code is available at
+    http://people.sc.fsu.edu/~jburkardt/py_src/sobol/sobol.html
+"""
+import math
+import sys
+import numpy as np
+
+# from numpy import np.bitwise_xor
+
+atmost = None
+dim_max = None
+dim_num_save = None
+initialized = None
+lastq = None
+log_max = None
+maxcol = None
+poly = None
+recipd = None
+seed_save = None
+v = None
+
+
+def i4_bit_hi1(n):
```

Comment:
what is "i4"?

Reply:
i4 means 4 bytes integer. This is related to the Sobol's algorithm which handle bits in a fancy way to get a low-discrepancy sequence of numbers.

Reply:
Could you please add a comment to explain it?